packages/opentelemetry-core/src/platform/browser/ShutdownNotifier.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+export function afterGlobalShutdown(cb: () => void): void {
```

Comment:
I would be in favour of removing this function and return a function which will do this when you are calling `onGlobalShutdown`. The reason is that people quite often use bind
```
onGlobalShutdown(myFunction.bind(this));
```
this means you cannot remove it unless you first create a reference to binded function
```
const myBindedFunction = myFunction.bind(this);
onGlobalShutdown(myBindedFunction);
afterGlobalShutdown(myBindedFunction);
```
this isn't very intuitive and error prone. 

Also the naming itself doesn't really explain what it does, should I call it always, or when, and what if I want to remove the event I just added etc.

Wheres if you define function to remove it when you actually calling onGlobalShutdown there is no way for user to make any mistake no matter what he do.

```
export function onGlobalShutdown(cb: () => void): void {
  window.addEventListener('unload', cb, {once: true});
  return function removeCallbackFromGlobalShutdown(){
    window.removeEventListener('unload', cb);
  }
}
```
so now it will look like this, simple straightforward, no option to make a mistake
```
const removeMe = onGlobalShutdown(myFunction.bind(this));
removeMe();
```

I'm thinking of more intuitive name maybe `NotifyOnGlobalShutDown(callback)` ?

Reply:
Currently, the only reason the functionality to remove the listener exists is for test cases. Once the graceful shutdown begins the event listener automatically cleans itself up, however in test cases this is not always triggered so I need to remove it between tests. Are we assuming users might want to use `onGlobalShutdown` themselves? I only intended for it to be used within `BasicTracerProvider` and `MeterProvider`. Also, I don't understand why a developer would want the event listener and the function which removes the event listener to use the same callback?

Reply:
we are exporting this, so any1 might be able to use it for many different purposes - that's why it needs to be solid, have docs etc.

Reply:
I agree with @obecny once you write a function you can't know how people will use it in the future so it should be well documented and solid.

Reply:
Ok, that makes sense to me. Could one of you comment on the same callback being used for the event listener and the function removing said event listener?

Reply:
I made some of these changes, please let me know how I can improve the implementation further.